mono/metadata/reflection.c

```diff
@@ -12308,7 +12387,7 @@ fix_partial_generic_class (MonoClass *klass)
 			}
 		} else {
 			mono_class_set_failure (klass, MONO_EXCEPTION_TYPE_LOAD, NULL);
-			mono_error_cleanup (&error);
+			mono_error_cleanup (error);
```

Comment:
This doesn't look like what we normally do.
We leave to the caller to decide what to do with the error.